tests/mock.rs

```diff
@@ -100,14 +99,14 @@ pub fn mock_serde_wallet<Rng: RngCore + CryptoRng>(
 /// Returns obfuscated notes with the given value.
 fn new_notes<Rng: RngCore + CryptoRng>(
     rng: &mut Rng,
-    psk: &PublicSpendKey,
+    pk: &PublicKey,
     note_values: &[u64],
 ) -> Vec<EnrichedNote> {
     note_values
         .iter()
         .map(|val| {
-            let blinder = JubJubScalar::random(rng);
-            (Note::new(rng, NoteType::Obfuscated, psk, *val, blinder), 0)
+            let blinder = JubJubScalar::random(&mut *rng);
```

Comment:
Isn't this just `rng`?
```suggestion
            let blinder = JubJubScalar::random(rng);
```

Reply:
if we use `rng`, the random method consumes it. This changed when we started using the upstream implementation of `JubJubScalar::random` which takes a `mut Rng` instead of a `&mut Rng`.